odl/tomo/data/mrc2014.py

```diff
@@ -0,0 +1,846 @@
+            # Need to invert the permutation
+            return tuple(ncells[ax]
+                         for ax in np.argsort(self.data_axis_order))
+
+    @property
+    def data_dtype(self):
+        """Data type of the data block as determined from `header`.
+
+        If the header entry ``'mode'`` is missing, ``None`` is returned.
+        Otherwise, it is determined from ``mode``.
+        """
+        if not self.header:
+            return None
+        try:
+            mode = int(self.header['mode']['value'])
+        except KeyError:
+            return None
+        else:
+            try:
+                return MRC_MODE_TO_NPY_DTYPE[mode]
+            except KeyError:
+                raise ValueError('data mode {} not supported'.format(mode))
+
+    @property
+    def data_axis_order(self):
+        """Permutation of ``(0, 1, 2)``.
+
+        If one of the header entries ``'mapc', 'mapr', 'maps'`` is missing,
+        the identity permutation ``(1, 2, 3)`` is returned.
+        Otherwise, value is determined from the ``'mapc', 'mapr', 'maps'``
+        header entries.
+        """
+        if not self.header:
+            return (0, 1, 2)
+        try:
+            mapc = self.header['mapc']['value']
+            mapr = self.header['mapr']['value']
+            maps = self.header['maps']['value']
+        except KeyError:
+            return (0, 1, 2)
+        else:
+            return tuple(int(m) - 1 for m in (mapc, mapr, maps))
+
+    def write_data(self, data, dstart=None, swap_axes=True):
+        """Write ``data`` to `file`.
+
+        Parameters
+        ----------
+        data : `array-like`
+            Data that should be written to `file`.
+        dstart : non-negative int, optional
+            Offset in bytes of the start position of the written data.
+            If provided, reshaping and axis swapping of ``data`` is
+            skipped.
+            For ``None``, `header_size` is used.
+        swap_axes : bool, optional
+            If ``True``, use the ``'mapc', 'mapr', 'maps'`` header entries
+            to swap the axes in the ``data`` before writing. Use ``False``
+            only if the data is already consistent with the final axis
+            order.
+        """
+        if dstart is None:
+            shape = self.data_shape
+            dstart = int(self.header_size)
+        elif dstart < 0:
+            raise ValueError('`dstart` must be non-negative, got {}'
+                             ''.format(dstart))
+        else:
+            shape = -1
+            dstart = int(dstart)
+
+        if dstart < self.header_size:
+            raise ValueError('invalid `dstart`, resulting in absolute '
+                             '`dstart` < `header_size` ({} < {})'
+                             ''.format(dstart, self.header_size))
+
+        data = np.asarray(data, dtype=self.data_dtype).reshape(shape)
+        if swap_axes:
+            data = np.transpose(data, axes=self.data_axis_order)
```

Comment:
Okay, I'll explain it to myself:
According to the documentation, we have the mapping
```python
num_cells[i] == data_shape[data_axis_order[i]]
```
which means that the x axis corresponds to the `data_axis_order[0]` axis in the data (and similar for y, z) *before transposing*, i.e. when reading raw from the file and reshaping into `data_shape`.
That means, in order to move the x axis into position 0, y to 1 and z to 2,  we need to permute the axes like in this example from the doc of `np.transpose`:
```python
>>> x = np.ones((1, 2, 3))
>>> np.transpose(x, (1, 0, 2)).shape
(2, 1, 3)
```
Since we need to transpose from `data_shape` to `num_cells`, we need the inverse behavior, since the argument to `transpose` is the "to" layout, you're right.
But then, the writer has to use the forward permutation instead. They can never be the same except for involutary permutations.